Lombiq.NodeJs.Extensions/scripts/validate-config.js

```diff
@@ -0,0 +1,89 @@
+/**
+ * @summary Validates the Node.js Extensions configuration object.
+ * @description This script validates a configuration object against the format provided in @example.
+ * @example "nodejsExtensions": {
+ *            "assetsToCopy": [
+ *              {
+ *                "sources": [ "Assets/Images" ],
+ *                "target": "wwwroot/images"
+ *              },
+ *              {
+ *                "sources": [
+ *                  "node_modules/marvelous/dist"
+ *                  "node_modules/wonderful/bin"
+ *                ],
+ *                "pattern": "*",
+ *                "target": "wwwroot"
+ *              }
+ *              // more groups, if needed
+ *            ],
+ *            "scripts": {
+ *              "source": "CustomJsFolder",
+ *              "target": "wwwCustomRoot/jay-es"
+ *            },
+ *            "styles": {
+ *              "source": "NonDefaultScssFolder",
+ *              "target": "wwwwNonDefaultCssFolder"
+ *            }
+ *          }
+ */
+function logError(message) { process.stderr.write(`ERROR: ${message}\n`); }
+
+function validateAssetGroupsAndLogErrors(assetConfig) {
+    if (!Array.isArray(assetConfig)) {
+        logError('The configuration must be an array of asset groups of the form: ' +
+            '{ sources: Array<string>, pattern: string [optional], target: string }.');
+        return false;
+    }
+
+    const errors = [];
+
+    assetConfig.forEach((assetGroup) => {
```

Comment:
Instead of `assetConfig.forEach` you can do `return assetConfig.reduce((success, assetGroup) => {  }, true)`. In the callback return `success` if there were no errors or `false` if there were. This way you don't need to maintain an external variable and you can return immediately.